pkg/operation/cloudbotanist/gcpbotanist/infrastructure.go

```diff
@@ -88,6 +117,44 @@ func (b *GCPBotanist) generateTerraformInfraConfig(createVPC bool, vpcName strin
 	}
 }
 
+func (b *GCPBotanist) destroyKubernetesFirewallRules() error {
+	t, err := b.NewShootTerraformer(common.TerraformerPurposeInfra)
+	if err != nil {
+		return err
+	}
+
+	if _, err := t.GetState(); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	vpcName := "vpc_name"
+	stateVariables, err := t.GetStateOutputVariables(vpcName)
+	if err != nil {
+		if terraformer.IsVariablesNotFoundError(err) {
+			b.Logger.Infof("Skipping explicit GCP load balancer and firewall rule deletion because not all variables have been found in the Terraform state.")
```

Comment:
```suggestion
			b.Logger.Infof("Skipping explicit GCP firewall rule deletion because not all variables have been found in the Terraform state.")
```